keras/src/models/model.py

```diff
@@ -399,6 +413,15 @@ def quantize(self, mode, **kwargs):
     def build_from_config(self, config):
         if not config:
             return
+        # Fetch the input structure from config if available.
```

Comment:
Lines 416-424 is all dead code. `"input_names"` is never in the config because nothing adds it there. Remove.

Coverage confirms it:
https://app.codecov.io/gh/keras-team/keras/pull/20842?src=pr&el=tree&filepath=keras%2Fsrc%2Fmodels%2Fmodel.py&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=keras-team#b343e88c9bde68f6f2b9d00f13fa225e-R417